pkg/controller/handlers/auditlogexport/auditlogexport.go

```diff
@@ -0,0 +1,240 @@
+	} else {
+		return fmt.Errorf("invalid storage config, no storage provider found")
+	}
+
+	// Create storage provider
+	storageProvider, err := auditlogexport.NewStorageProvider(provider, h.credProvider)
+	if err != nil {
+		return fmt.Errorf("failed to create storage provider: %w", err)
+	}
+
+	export.Status.StorageProvider = provider
+
+	// Generate export path
+	exportPath := h.generateExportPath(export)
+
+	// Use streaming export with batching
+	_, exportSize, err := h.streamingExport(ctx, export, storageProvider, exportPath)
+	if err != nil {
+		return fmt.Errorf("failed to perform streaming export: %w", err)
+	}
+
+	// Update export status with results
+	export.Status.ExportSize = exportSize
+	export.Status.ExportPath = exportPath
+	export.Status.State = types.AuditLogExportStateCompleted
+	export.Status.CompletedAt = &metav1.Time{Time: time.Now()}
+
+	return nil
+}
+
+func (h *Handler) streamingExport(ctx context.Context, export *v1.AuditLogExport, storageProvider auditlogexport.StorageProvider, exportPath string) (int64, int64, error) {
+	storageConfig, err := h.credProvider.GetStorageConfig(ctx)
+	if err != nil {
+		return 0, 0, fmt.Errorf("failed to get storage config: %w", err)
+	}
+
+	const batchSize = 10000 // Process 10,000 records per batch
+
+	var totalRecords int64
+	var totalSize int64
+	offset := 0
+	batchNumber := 0
+
+	pr, pw := io.Pipe()
+	defer pr.Close()
+
+	uploadErrCh := make(chan error, 1)
+	go func() {
+		defer close(uploadErrCh)
+		err := storageProvider.Upload(ctx, *storageConfig, export.Spec.Bucket, exportPath, pr)
+		uploadErrCh <- err
+	}()
+
+	for {
+		// Prepare batch options
+		opts := client.MCPAuditLogOptions{
+			StartTime:                 export.Spec.StartTime.Time,
+			EndTime:                   export.Spec.EndTime.Time,
+			UserID:                    export.Spec.Filters.UserIDs,
+			MCPID:                     export.Spec.Filters.MCPIDs,
+			MCPServerDisplayName:      export.Spec.Filters.MCPServerDisplayNames,
+			MCPServerCatalogEntryName: export.Spec.Filters.MCPServerCatalogEntryNames,
+			CallType:                  export.Spec.Filters.CallTypes,
+			CallIdentifier:            export.Spec.Filters.CallIdentifiers,
+			SessionID:                 export.Spec.Filters.SessionIDs,
+			ClientName:                export.Spec.Filters.ClientNames,
+			ClientVersion:             export.Spec.Filters.ClientVersions,
+			ResponseStatus:            export.Spec.Filters.ResponseStatuses,
+			ClientIP:                  export.Spec.Filters.ClientIPs,
+			Query:                     export.Spec.Filters.Query,
+			Limit:                     batchSize,
+			Offset:                    offset,
+			WithRequestAndResponse:    export.Spec.WithRequestAndResponse,
+		}
+
+		// Get batch of logs from gateway
+		logs, total, err := h.gatewayClient.GetMCPAuditLogs(ctx, opts)
+		if err != nil {
+			return 0, 0, fmt.Errorf("failed to get audit logs batch %d: %w", batchNumber, err)
+		}
+
+		totalRecords = total
```

Comment:
```suggestion
		totalRecords += total
```

Reply:
this shoud be correct, the total is the total record that you will get regardless of limit and offset.